### PR TITLE
[GOOWOO-471] Limit image format selection in campaign asset creation

### PR DESCRIPTION
**Pull request by AlejandroPerezMartin**
### Changes proposed in this Pull Request:

Closes https://linear.app/a8c/issue/GOOWOO-471/addcheck-support-for-webp-image-format-in-campaign-asset-creation

Restricts the usage of non supported formats to the campaign asset creation flow.

### Screenshots:

Error message:
Screenshot 2026-03-19 at 09 43 53

### Detailed test instructions:

1. Create a new campaign
2. On the assets section, click on "Add image +"
3. Ensure that only `.png`, `.jpg`or `.gif` images can be selected from the media library
4. When uploading a non supported format, like `.webp` it should display a notice error next to the "Select" button in the bottom right corner of the modal and the button to select the image should also be disabled

### Changelog entry

> Tweak - Restrict image formats in campaign asset creating

### CODE REVIEW

File: js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.js [hunk 1]
```diff
@@ -4,6 +4,17 @@
 import { __, sprintf } from '@wordpress/i18n';
 import { useCallback, useRef } from '@wordpress/element';
 
+/**
+ * The MIME types allowed for campaign asset image selection.
+ * WebP and other formats are not supported.
+ */
+export const ALLOWED_MIME_TYPES = [
```
Comment (asvinb) [inline js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.js]: @AlejandroPerezMartin Why are we exporting that const? 🤔

File: js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.js [hunk 2]
```diff
@@ -4,6 +4,17 @@
 import { __, sprintf } from '@wordpress/i18n';
 import { useCallback, useRef } from '@wordpress/element';
 
+/**
```
Comment (asvinb) [inline js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.js:12]: @AlejandroPerezMartin This is should after the Internal dependencies comment please.

File: js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.js [hunk 3]
```diff
@@ -322,17 +348,26 @@ export default function useCroppedImageSelector( {
...
+				let invalidMimeType;
 
 				if ( selection.length ) {
-					const { width, height } = selection.first().toJSON();
+					const { width, height, mime } = selection.first().toJSON();
 					invalidSize = width < minWidth || height < minHeight;
+					invalidMimeType = ! isMimeTypeAllowed( mime );
+				}
+
+				if ( invalidSize ) {
```
Comment (asvinb) [inline js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.js]: @AlejandroPerezMartin Let's move the `invalidSize` and `invalidMimeType` checks inside the `selection.length` block. That way we can set `errorMessage` directly, removing the need for the intermediate `invalidSize` and `invalidMimeType` variables.

File: js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.test.js [hunk 4]
```diff
@@ -71,3 +72,16 @@ describe( 'calcRatioPercentError', () => {
 		expect( calcRatioPercentError( 200, 100, 1, 1 ) ).toBeCloseTo( 100 );
 	} );
 } );
+
+describe( 'isMimeTypeAllowed', () => {
```
Comment (asvinb) [inline js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.test.js:76]: @AlejandroPerezMartin We can simplify this test as such:
```
test.each( [
    [ 'image/jpg',  true  ],
    [ 'image/jpeg', true  ],
    [ 'image/png',  true  ],
    [ 'image/gif',  true  ],
    [ 'image/webp', false ],
] )( 'isMimeTypeAllowed(%s) should return %s', ( mime, expected ) => {
    expect( isMimeTypeAllowed( mime ) ).toBe( expected );
} );
```

**Review by asvinb — CHANGES_REQUESTED** [review]

File: js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.js (hunk 1 above)
Comment (AlejandroPerezMartin) [reply js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.js]: removed the export

File: js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.test.js (hunk 4 above)
Comment (AlejandroPerezMartin) [reply js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.test.js:76]: nice, applied!

File: js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.js (hunk 3 above)
Comment (AlejandroPerezMartin) [reply js/src/hooks/useCroppedImageSelector/useCroppedImageSelector.js]: done!

**Review by asvinb — APPROVED** [review]
LGTM @AlejandroPerezMartin !

File: js/src/pages/edit-paid-ads-campaign/index.js [hunk 5]
```diff
@@ -219,6 +219,7 @@ const EditPaidAdsCampaign = () => {
 
 		getHistory().push( getDashboardUrl() );
 	};
+	console.log( 'campaign', campaign );
```
Comment (AlejandroPerezMartin) [inline js/src/pages/edit-paid-ads-campaign/index.js]: let's remove the log

**Comment by ankitguptaindia** [comment]
QA Status: Verfied ✅ 
---- 

image


## Test Summary

- Feature: Limit image format selection in **campaign asset creation** and **campaign asset editing** so that only supported formats can be used when adding images to paid ads campaigns.   
- Status: **Passed** – all described scenarios were tested and are working as expected.
- Scope: Media library selection filter in the campaign asset flow, validation of uploaded files (including `.webp`), inline error messaging, and disabling of the Select button for unsupported formats.   

***

## Test Environment

- Branch: `add/GOOWOO-471-genai-webp-image-check`
- Area: Campaign creation flow → Assets section → “Add image +”
***

## Test Scenarios and Results

### 1. Supported formats from the media library

- Steps:  
  - Create a new campaign.  
  - In the Assets section, click **Add image +** to open the media modal.  
  - Select images with extensions `.png`, `.jpg`/`.jpeg`, and `.gif`.   
- Expected:  
  - Only `.png`, `.jpg`/`.jpeg`, and `.gif` are available/selectable as valid formats.   
  - Select button is enabled for these images and allows confirming the selection.   
- Result: **Pass** – supported formats can be selected and used without errors, and no unsupported formats appear as valid choices.   

### 2. Uploading an unsupported format (e.g., .webp)

- Steps:  
  - In the same media modal, upload a `.webp` image (or another unsupported format) via the upload tab.   
- Expected:  
  - An inline **error notice** is displayed next to the Select button at the bottom right of the modal, indicating the format is not supported.   
  - The Select button for that image is disabled so the asset cannot be used in the campaign.   
- Result: **Pass** – the error message appears as in the PR screenshot, and the Select button is disabled for unsupported formats.   



### 3. Error state behavior and UX

- Steps:  
  - Trigger the error by selecting or uploading a `.webp` image.  
  - Observe the error copy, placement, and any state reset when changing selection back to a supported image.   
- Expected:  
  - Error text is visible and clearly communicates that only `.png`, `.jpg`/`.jpeg`, and `.gif` formats are supported.   
  - When a supported image is selected instead, the error state is cleared and the Select button is re‑enabled.   
- Result: **Pass** – error copy and behavior match the PR description, and the flow recovers correctly after switching to a valid image.   

### 4. Campaign creation end‑to‑end

- Steps:  
  - Create a new campaign, add at least one supported image via **Add image +**, complete the remaining setup, and save the campaign.   
- Expected:  
  - Campaign can be created successfully using supported image formats as assets.  
  - No regressions in the rest of the campaign setup flow.   
- Result: **Pass** – campaigns using supported formats are created as expected; no new errors observed in adjacent steps.   

***

## Final QA Status

- Overall result: **Passed** – all available test cases from the PR description and implied edge cases (supported vs unsupported formats) are working as expected.   
- QA feedback: No outstanding issues; behavior aligns with the stated goal of restricting campaign asset images to `.png`, `.jpg`/`.jpeg`, and `.gif` formats and preventing use of `.webp` and other unsupported formats.   
- Recommendation: Ready to merge once CI checks remain green.

**Review by ankitguptaindia — APPROVED** [review]

**Comment by ankitguptaindia** [comment]
@AlejandroPerezMartin Could you please resolve the merge conflicts here? thanks! 

cc: @asvinb
